File: go/guessing_game.go
package main

import (
	"bufio"     // for the scanner
	"fmt"       // for println
	"math/rand" // for rng
	"os"        // for stdin
	"strconv"   // for converting strings to integers
	"strings"   // for trimming
	"time"      // for seeding our rng
)

func main() {
	scanner := bufio.NewScanner(os.Stdin)
	rng := rand.NewSource(time.Now().UnixNano())

	for {
		fmt.Println("Guess a number between 1 - 100!")

		target := int(rng.Int63()%100 + 1)
		guess := -1 // shorthand for 'var guess = 1;'

		for target != guess {
			fmt.Print("Guess: ")
			scanner.Scan()
			input, err := strconv.Atoi(strings.TrimSpace(scanner.Text()))

			if err != nil {
				fmt.Println("Please type a number!")
				continue
			}
			guess = input

			if guess < target {
				fmt.Println("Higher!")
			} else if guess > target {
				fmt.Println("Lower!")
			}
		}
		fmt.Println("You guessed it!")

		fmt.Println("Play again?")
		fmt.Print("(Y/n): ")
		scanner.Scan()
		if strings.HasPrefix(strings.ToLower(strings.TrimSpace(scanner.Text())), "n") {
			break
		}
	}
}

File: java/GuessingGame.java
import java.util.*;
import java.lang.*;

public class GuessingGame {
    public static void main(String... args) {
        Random rng = new Random();
        Scanner scanner = new Scanner(System.in);

        do {
            int target = 1 + rng.nextInt(100);
            int guess = -1;
            System.out.println("Guess a number between 1 - 100!");
    
            while (target != guess) {
                try {
                    System.out.print("Guess: ");
                    guess = Integer.parseInt(scanner.nextLine());
                } catch (NumberFormatException e) {
                    System.out.println("Please type a number!");
                    continue;
                }
    
                if      (guess < target) { System.out.println("Higher!"); }
                else if (guess > target) { System.out.println("Lower!"); }
                else {
                    System.out.println("You guessed it!");
                }
            }
            System.out.println("Play again?");
            System.out.print("(Y/n): ");
        } while (scanner.nextLine().trim().toLowerCase().charAt(0) != 'n');


        scanner.close(); // This is not nessecary but it makes VSCode happy.
        // I hope you're happy now Code, you better be.
    }
}